File: api/src/main/java/com/bartlomiejpluta/base/api/map/layer/color/ColorLayer.java
package com.bartlomiejpluta.base.api.map.layer.color;

import com.bartlomiejpluta.base.api.map.layer.base.Layer;
import com.bartlomiejpluta.base.internal.object.Placeable;
import com.bartlomiejpluta.base.internal.render.Renderable;

public interface ColorLayer extends Placeable, Renderable, Layer {
   void setColor(float red, float green, float blue, float alpha);

   void setColor(float red, float green, float blue);

   void setRed(float red);

   void setGreen(float green);

   void setBlue(float blue);

   void setAlpha(float alpha);
}

File: editor/src/main/kotlin/com/bartlomiejpluta/base/editor/map/model/layer/ColorLayer.kt
package com.bartlomiejpluta.base.editor.map.model.layer

import javafx.beans.property.SimpleObjectProperty
import javafx.beans.property.SimpleStringProperty
import tornadofx.getValue
import tornadofx.setValue

class ColorLayer(name: String, red: Int, green: Int, blue: Int, alpha: Int) : Layer {
   override val nameProperty = SimpleStringProperty(name)

   override var name by nameProperty

   val redProperty = SimpleObjectProperty(red)
   var red by redProperty

   val greenProperty = SimpleObjectProperty(green)
   var green by greenProperty

   val blueProperty = SimpleObjectProperty(blue)
   var blue by blueProperty

   val alphaProperty = SimpleObjectProperty(alpha)
   var alpha by alphaProperty

   override fun resize(rows: Int, columns: Int) {
      // We essentially need to do nothing
   }
}